webui/react/src/hooks/useModal/useModalChangePassword.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button, Form, Input } from 'antd';
+import React from 'react';
+
+import { useStore } from 'contexts/Store';
+import { login, setUserPassword } from 'services/api';
+
+import useModal, { ModalHooks } from './useModal';
+import css from './useModalChangePassword.module.scss';
+
+const useModalChangePassword = (): ModalHooks => {
+  const { modalClose, modalOpen: openOrUpdate, modalRef } = useModal();
+  const { auth } = useStore();
+  const username = auth.user?.username || 'Anonymous';
+  const [ form ] = Form.useForm();
+
+  const getModalContent = () => {
+    const submitForm = async (): Promise<void> => {
+      await setUserPassword({
+        password: form.getFieldValue('newPassword'),
+        username,
+      });
+      modalClose();
+    };
```

Comment:
a couple of things:
1. why is `submitForm` defined inside `getModalContent` instead of just being a top level `useCallback` function inside of `useModalChangePassword`?
2. `await setUserPassword` should have a try/catch around it to handle any errors